libs/pika/schedulers/include/pika/schedulers/shared_priority_queue_scheduler.hpp

```diff
@@ -647,25 +639,28 @@ namespace pika { namespace threads { namespace policies {
             // if we did not get a task at all, then try converting
             // tasks in the pending queue into staged ones
             std::size_t added = 0;
-            std::int64_t idle_loop_count = 0;
-            wait_or_add_new(thread_num, true, idle_loop_count, true, added);
+            just_add_new(added);
             if (added > 0)
-                return get_next_thread(
-                    this_thread, running, thrd, enable_stealing);
+                return get_next_thread(0, true, thrd, enable_stealing);
             return false;
         }
 
-        /// Return the next thread to be executed, return false if none available
+        // just a dummy function to keep the scheduling loop happy
```

Comment:
This is not completely dummy, right? It just doesn't do the `wait` part? Though I'm not even sure the `wait` in `wait_or_add_new` makes sense for other schedulers either...